azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/DeploymentToNodeMetadata.java

```diff
@@ -0,0 +1,162 @@
+   //
+   // To get details about the resource(s) deployed via template, one needs to query the
+   // various resources after the deployment has "SUCCEEDED".
+   private static final Map<Deployment.ProvisioningState, NodeMetadata.Status> STATUS_TO_NODESTATUS =
+           ImmutableMap.<Deployment.ProvisioningState, NodeMetadata.Status>builder().
+                   put(Deployment.ProvisioningState.ACCEPTED, NodeMetadata.Status.PENDING).
+                   put(Deployment.ProvisioningState.READY, NodeMetadata.Status.PENDING).
+                   put(Deployment.ProvisioningState.RUNNING, NodeMetadata.Status.PENDING).
+                   put(Deployment.ProvisioningState.CANCELED, NodeMetadata.Status.TERMINATED).
+                   put(Deployment.ProvisioningState.FAILED, NodeMetadata.Status.ERROR).
+                   put(Deployment.ProvisioningState.DELETED, NodeMetadata.Status.TERMINATED).
+                   put(Deployment.ProvisioningState.SUCCEEDED, NodeMetadata.Status.RUNNING).
+                   put(Deployment.ProvisioningState.UNRECOGNIZED, NodeMetadata.Status.UNRECOGNIZED).
+                   build();
+
+   public static Deployment.ProvisioningState provisioningStateFromString(final String text) {
+      if (text != null) {
+         for (Deployment.ProvisioningState status : Deployment.ProvisioningState.values()) {
+            if (text.equalsIgnoreCase(status.name())) {
+               return status;
+            }
+         }
+      }
+      return Deployment.ProvisioningState.UNRECOGNIZED;
+   }
+
+   private final AzureComputeApi api;
+
+   private final Supplier<Set<? extends Location>> locations;
+
+   private final GroupNamingConvention nodeNamingConvention;
+
+   private final VMImageToImage vmImageToImage;
+
+   private final VMHardwareToHardware vmHardwareToHardware;
+
+   private final Map<String, Credentials> credentialStore;
+
+   @Inject
+   DeploymentToNodeMetadata(
+           AzureComputeApi api,
+           @Memoized Supplier<Set<? extends Location>> locations,
+           GroupNamingConvention.Factory namingConvention, VMImageToImage vmImageToImage,
+           VMHardwareToHardware vmHardwareToHardware, Map<String, Credentials> credentialStore) {
+
+      this.nodeNamingConvention = namingConvention.createWithoutPrefix();
+      this.locations = locations;
+      this.vmImageToImage = vmImageToImage;
+      this.vmHardwareToHardware = vmHardwareToHardware;
+      this.credentialStore = credentialStore;
+      this.api = api;
+   }
+
+   @Override
+   public NodeMetadata apply(final VMDeployment from) {
+      final NodeMetadataBuilder builder = new NodeMetadataBuilder();
+      final Deployment deployment = from.deployment;
+      builder.id(deployment.name());
+      builder.providerId(deployment.name());
+      builder.name(deployment.name());
+      String group = deployment.name();
+      int index = group.lastIndexOf("-");
+      builder.group(group.substring(0, index));
```

Comment:
If this is parsing a jclouds generated name, then the group naming convention already has methods to get the group part. Use that, since the separator used by the convention is also configurable by properties.